lib/matplotlib/ticker.py

```diff
@@ -2092,27 +2092,27 @@ def _raw_ticks(self, vmin, vmax):
         scale, offset = scale_range(vmin, vmax, nbins)
         _vmin = vmin - offset
         _vmax = vmax - offset
-        raw_step = (_vmax - _vmin) / nbins
         steps = self._extended_steps * scale
         if self._integer:
             # For steps > 1, keep only integer values.
             igood = (steps < 1) | (np.abs(steps - np.round(steps)) < 0.001)
             steps = steps[igood]
 
-        istep = np.nonzero(steps >= raw_step)[0][0]
-
-        # Classic round_numbers mode may require a larger step.
+        good_steps = steps >= ((_vmax - _vmin) / nbins)
```

Comment:
If I follow the code correctly (perhaps I'm not) at that point the steps will be [bad0, bad1, ..., badN, good0, good1, ..., goodN] but actually the steps that will be considered below will be (in that order) good0, badN, bad(N-1), ... and good1, ... will never be considered?  In which case I'd say the name "good_steps" is a bit confusing.